icinga2/util.go

```diff
@@ -0,0 +1,49 @@
+package icinga2
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+func (s Service) MarshalJSON() ([]byte, error) {
+	// Prevent json.Marshal() recursion
+	type service Service
+	serviceAsJson, err := json.Marshal(service(s))
```

Comment:
This should be as follows so that we can still unmarshal `Vars` using the default json unmarshalling:

```suggestion
	svc := service(s)
	// Clear top-level Vars field so it's not added to the marshalled JSON. We marshal all service variables into individual top-level `vars.<variable name>` fields below.
	svc.Vars = Vars{}
	serviceAsJson, err := json.Marshal(svc)
```